xdsl/dialects/riscv_scf.py

```diff
@@ -74,10 +74,63 @@ def __init__(
         )
 
 
+@irdl_op_definition
+class WhileOp(IRDLOperation):
+    name = "riscv_scf.while"
+    arguments: VarOperand = var_operand_def(IntRegisterType)
```

Comment:
`ForOp` uses `RISCVRegisterType` for arguments (so either float or int), we should probably do the same here to be future proof

Reply:
I'm not sure we allow float for `ForOp`? It's kind of weird.

Reply:
RISCVRegisterType is the parent class for both, so we do allow in fact.
The only limitation is the current lowering, but that'll be also changed in the nearby future.